app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
+
+  def create
+    if generate_post
+      redirect_to create_success_path
+    else
+      redirect_to create_fail_path
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @post.update_attributes(posts_params)
+      redirect_to update_success_path
+    else
+      redirect_to update_fail_path
+    end
+  end
+
+  def destroy
+    if @post.destroy
+      redirect_to destroy_success_path
+    else
+      redirect_to destroy_fail_path
+    end
+  end
+
+  def reply
+    @post = @topic.posts.build(posts_params)
+    if @post.save
+      redirect_to reply_success_path
+    else
+      redirect_to reply_fail_path
+    end
+  end
+
+  def default_return_path
+    fail NotImplementedError, 'To be implemented by the specific topic posts controller.'
+  end
+
+  def auto_subscribe_topic
+    fail NotImplementedError, 'To be implemented by the specific topic posts controller.'
+  end
+
+  def topic
+    fail NotImplementedError, 'To be implemented by the specific topic posts controller.'
+  end
+
+  protected
+
+  def create_success_path
+    default_return_path
+  end
+
+  def create_fail_path
+    default_return_path
+  end
+
+  def update_success_path
+    default_return_path
+  end
+
+  def update_fail_path
+    default_return_path
+  end
+
+  def destroy_success_path
+    default_return_path
+  end
+
+  def destroy_fail_path
+    default_return_path
+  end
+
+  def reply_success_path
+    default_return_path
+  end
+
+  def reply_fail_path
+    default_return_path
+  end
+
+  private
+
+  def set_topic
+    @topic ||= topic
+  end
+
+  def generate_post
+    @post ||= @topic.posts.build(posts_params)
```

Comment:
Then why we need `Discussion::Topic`? Does anything that can be commented act as a `Discussion::Topic`?